gelfformatter/formatter.py

```diff
@@ -89,12 +89,17 @@ class GelfFormatter(logging.Formatter):
        https://docs.python.org/3/library/logging.html#logrecord-attributes
     """
 
-    def __init__(self, allowed_reserved_attrs=[]):
+    def __init__(self, allowed_reserved_attrs=[], additional_reserved_attrs=[]):
         """Initializes a GelfFormatter."""
         super(GelfFormatter, self).__init__()
         self.allowed_reserved_attrs = allowed_reserved_attrs
         self._hostname = socket.gethostname()
 
+        self._reserved_attrs = [
```

Comment:
Following the comment above, I think we should expose a new instance attribute to the outside, named `ignored_attrs`, similar to `allowed_reserved_attrs`. We may want to change the list of ignored fields dynamically on the fly (this is why I build the `exclude` list on the `format` method instead of `__init__`, so we'll need to keep the same behavior here).

Considering the above, we should just assign the input `ignored_attrs` to `self.ignored_attrs`, and then we can compute the full list of fields to exclude based on `allowed_reserved_attrs`, `RESERVED_ATTRS` and `ignored_attrs` in the `format` method.

We should also document this new attribute in the class doctoring.